### PR TITLE
Main basic framework nearly complete

### PR DESCRIPTION
**Pull request by rickboot**
Done:

Main page layout
PropTyping
CSS Modules

To do:
- # Answers mini component
- Next/prev page
- Make answers not display if 0
- Sort by (real site is broken) 
- dynamic forms for ask and answer
- mouseover popup for user
- top nav click
- CSS dim and time animations for buttons

### CODE REVIEW

File: client/src/components/App.css [hunk 1]
```diff
@@ -1,4 +1,3 @@
 .container {
   display: flex;
-  justify-content: center;
-}
+  justify-content: center;
```
Comment (ryanminjaeshin) [inline client/src/components/App.css:3]: I am going to reference some of your css @rickboot 👍

File: client/src/components/Questions/Questions.jsx [hunk 2]
```diff
@@ -29,8 +29,8 @@ class Questions extends React.Component {
         this.setState({
```
Comment (ryanminjaeshin) [inline client/src/components/Questions/Questions.jsx:29]: Maybe you want to use callback?

File: client/src/components/Questions/QuestionsList/Answer/Answer.jsx [hunk 3]
```diff
@@ -1,23 +1,58 @@
...
+import Button from './Button/Button';
+
 import style from './Answer.css';
 
 function Answer({ answer }) {
   return (
     <div className={style.answer}>
-      <span>{answer.text}</span>
+      <div className={style.user}>
+        <span>{`${answer.user.nickname} · x days ago`}</span>
```
Comment (ryanminjaeshin) [inline client/src/components/Questions/QuestionsList/Answer/Answer.jsx:11]: It seems like you are planning to change 'x' to mock data later. 👍

File: client/src/components/Questions/QuestionsList/Answer/Answer.jsx [hunk 4]
```diff
@@ -1,23 +1,58 @@
...
+        <Button text="Report" handleClick={() => {}} />
+      </div>
     </div>
   );
 }
 
 Answer.propTypes = {
-  answer: PropTypes.object,
-  // answer.text: PropTypes.string,
+  answer: PropTypes.shape({
```
Comment (ryanminjaeshin) [inline client/src/components/Questions/QuestionsList/Answer/Answer.jsx:29]: lol I had to do these too.

File: client/src/components/Questions/QuestionsList/Answer/Button/Button.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+      type="button"
+      onClick={handleClick}
+      className={style.button}
+    >
+      <span>{text}</span>
+    </button>
+  );
+}
+
+Button.propTypes = {
```
Comment (ryanminjaeshin) [inline client/src/components/Questions/QuestionsList/Answer/Button/Button.jsx:17]: Do you also need defaultProps here?

Comment (rickboot) [reply client/src/components/Questions/QuestionsList/Answer/Button/Button.jsx:17]: It is marked 'isRequired' so there is no need for default

Comment (rickboot) [reply client/src/components/Questions/QuestionsList/Answer/Button/Button.jsx:17]: It is marked 'isRequired' so no need for defaultProps

**Review by ryanminjaeshin — APPROVED** [review]
All the codes look good to me. I hope you resolve the git workflow issue 🥇

**Pull request merged**